internal/server/grpc.go

```diff
@@ -0,0 +1,115 @@
+package server
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"log/slog"
+	"net"
+	"time"
+
+	"github.com/supressionstop/xenking_test_1/internal/server/pb"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/peer"
+	"google.golang.org/grpc/status"
+)
+
+type Client string
+
+type Grpc struct {
+	addr                string
+	grpcServer          *grpc.Server
+	logger              *slog.Logger
+	subscriptionManager *SubscriptionManager
+	pb.UnimplementedLinesServer
```

Comment:
In the current implementation, this `server` package serves as a transport and controller handler. I believe this should be separated into a transport level (HTTP, grpc, or Async) and an actual controller level where we will use `SubscriptionManager`, `worker.Pool` etc. At a transport level, we should apply the ports approach to provide all user inputs to the controller level.

Reply:
TODO:
1. extract gRPC and HTTP into separate package - transport;
2. extract transport controllers into separate package - controllers